components/match2/wikis/trackmania/match_group_input_custom.lua

```diff
@@ -0,0 +1,538 @@
+			match['map' .. index] = map
+		end
+	end
+	return match
+end
+
+function matchFunctions.getExtraData(match)
+	local opponent1 = match.opponent1 or {}
+	local opponent2 = match.opponent2 or {}
+
+	local casters = {}
+	for key, name in Table.iter.pairsByPrefix(match, 'caster') do
+		table.insert(casters, CustomMatchGroupInput._getCasterInformation(
+			name,
+			match[key .. 'flag'],
+			match[key .. 'name']
+		))
+	end
+	table.sort(casters, function(c1, c2) return c1.displayName:lower() < c2.displayName:lower() end)
+
+	local showh2h = Logic.readBool(match.showh2h)
+		and opponent1.type == Opponent.team
+		and opponent2.type == Opponent.team
+
+	match.extradata = {
+		team1icon = getIconName(opponent1.template or ''),
+		team2icon = getIconName(opponent2.template or ''),
+		lastgame = Variables.varDefault('last_game'),
+		isconverted = 0,
+		showh2h = showh2h,
+		isfeatured = matchFunctions.isFeatured(match),
+		casters = Table.isNotEmpty(casters) and Json.stringify(casters) or nil,
+	}
+	return match
+end
+
+function CustomMatchGroupInput._getCasterInformation(name, flag, displayName)
+	if String.isEmpty(flag) then
+		flag = Variables.varDefault(name .. '_flag')
+	end
+	if String.isEmpty(displayName) then
+		displayName = Variables.varDefault(name .. '_dn')
+	end
+	if String.isEmpty(flag) or String.isEmpty(displayName) then
+		local parent = Variables.varDefault(
+			'tournament_parent',
+			mw.title.getCurrentTitle().text
+		)
+		local pageName = mw.ext.TeamLiquidIntegration.resolve_redirect(name)
+		local data = mw.ext.LiquipediaDB.lpdb('broadcasters', {
+			conditions = '[[page::' .. pageName .. ']] AND [[parent::' .. parent .. ']]',
+			query = 'flag, id',
+			limit = 1,
+		})
+		if type(data) == 'table' and data[1] then
+			flag = String.isNotEmpty(flag) and flag or data[1].flag
+			displayName = String.isNotEmpty(displayName) and displayName or data[1].id
+		end
+	end
+	if String.isNotEmpty(flag) then
+		Variables.varDefine(name .. '_flag', flag)
+	end
+	if String.isEmpty(displayName) then
+		displayName = name
+	end
+	if String.isNotEmpty(displayName) then
+		Variables.varDefine(name .. '_dn', displayName)
+	end
+	return {
+		name = name,
+		displayName = displayName,
+		flag = flag,
+	}
+end
+
+function matchFunctions.isFeatured(match)
+	local opponent1 = match.opponent1
+	local opponent2 = match.opponent2
+	if opponent1.type ~= 'team' or opponent2.type ~= 'team' then
+		return false
+	end
+
+	if
+		tonumber(match.liquipediatier or '') == 1
+		or tonumber(match.liquipediatier or '') == 2
+		or Logic.readBool(Variables.varDefault('ubisoftmajor'))
+		or not String.isEmpty(Variables.varDefault('match_featured_override'))
+	then
+		return true
+	end
+
+	if matchFunctions.currentEarnings(opponent1.name) >= _EARNINGS_LIMIT_FOR_FEATURED then
+		return true
+	elseif matchFunctions.currentEarnings(opponent2.name) >= _EARNINGS_LIMIT_FOR_FEATURED then
+		return true
+	end
+	return false
+end
+
+function matchFunctions.currentEarnings(name)
```

Comment:
if we keep this i suggest to cache the values in wikivars and reuse them instead of querying the same data for the same team over and over again

Reply:
https://github.com/Liquipedia/Lua-Modules/pull/2259/commits/d07b807243a568a3bd1cda2add3c871166543bfc